api/proto/config/storage.proto

```diff
@@ -34,13 +35,18 @@ message DatastoreConfig {
 
 	string apiKey = 3;
 
-	string password = 4;
+    // to be decrypted, use cipherTextPassword instead
+    string password = 4;
 
 	// depending on datastore types, it could be bucket or path
 	string dpath = 5;
 
 	// Applies for some datastore types
 	string region = 6;
+
+    bytes cipherTextPassword = 7;
+
+    CipherInfo CInfo = 8;
```

Comment:
Same comment here; CipherInfo cipherTextPassword

But we might want to protect the apiKey as well, which means that we should put apiKey+password in a message and encrypt that message.